src/python/pants/backend/shell/shell_command.py

```diff
@@ -140,7 +141,7 @@ async def prepare_shell_command_process(
                 binary_name=tool,
                 search_path=search_path,
             )
-            for tool in (*tools, *["mkdir", "ln"])
+            for tool in OrderedSet(*tools, *["mkdir", "ln"])
```

Comment:
It doesn't really matter - don't waste CI. More an FYI: `FrozenOrderedSet` is often preferable because it's immutable.

Reply:
Well, this shouldn't rely on *tools ordering anyhow really. If a user swaps "foo", "bar" to "bar", "foo", this will all re-run and it needn't. Probably best to dedup + sort.

Reply:
We are very much in bikeshed territory, so please don't change it again, but: sorting and then dedup'ing is more expensive than an unordered dedup'ing with a set (from a big `O` notation perspective), so I wouldn't call "prefer sorting and dedup'ing" good general advice.

And FrozenOrderedSet really doesn't matter in this position: it's being consumed immediately.

Reply:
FWIW I landed on dedup-then-sort and use a vanilla `set` for deduping :smile: